AmplifyTools/amplify-tools.sh

```diff
@@ -6,13 +6,121 @@
 # SPDX-License-Identifier: Apache-2.0
 
 set -e
-export PATH=$PATH:$(npm bin -g)
 
 if ! which node >/dev/null; then
   echo "warning: Node is not installed. Visit https://nodejs.org/en/download/ to install it"
   exit 1
-elif ! test -f ./amplifytools.xcconfig; then
-  npx amplify-app --platform ios
+fi
+
+export PATH=$PATH:$(npm bin -g)
+
+# Note the use of tail -1 is important here because when upgrading between versions
+# the first time that you run these commands, we have seen this variable take on the value of:
+# """
+# Scanning for plugins...
+# plugin scan successful
+# 4.21.0
+# """
+AMP_APP_VERSION_CURRENT=`npx -q amplify-app --version |tail -1`
+AMP_APP_VERSION_MINIMUM="2.17.1"
+AMP_APP_VERSION_INVALID=0
+
+AMP_CLI_VERSION_CURRENT=`npx -q amplify --version |tail -1`
+AMP_CLI_VERSION_MINIMUM="4.21.0"
+AMP_CLI_VERSION_INVALID=0
+
+
+STRIP_ESCAPE_RESULT=
+stripEscapeUtil() {
+    STRIP_ESCAPE_RESULT=
+
+    set +e
+    HAS_RUBY=0
+    which ruby > /dev/null 2>&1
+    if [[ $? -eq 0 ]]; then
+	HAS_RUBY=1
+    fi
+    set -e
+
+    if [ ${HAS_RUBY} -eq 1 ]; then
+	STRIP_ESCAPE_RESULT=`echo "${1}" | ruby -pe 'gsub(/\e\[[0-9;]*m/s, "")'`
+    else
+	STRIP_ESCAPE_RESULT=`echo "${1}" | npx -q strip-ansi-cli`
+    fi
+}
+
+VERSION_INVALID=
+checkMinVersionCompatibility() {
+    VERSION_INVALID=0
+
+    stripEscapeUtil "${1}"
+    CURR_VERSION_SAFE="${STRIP_ESCAPE_RESULT}"
+    CURR_VERSION_MAJOR=`echo "${CURR_VERSION_SAFE}" | tr  '.' ' ' | awk '{print $1}'`
+    CURR_VERSION_MINOR=`echo "${CURR_VERSION_SAFE}" | tr  '.' ' ' | awk '{print $2}'`
+    CURR_VERSION_PATCH=`echo "${CURR_VERSION_SAFE}" | tr  '.' ' ' | awk '{print $3}'`
+
+    stripEscapeUtil "${2}"
+    REQU_VERSION_SAFE="${STRIP_ESCAPE_RESULT}"
+    REQU_VERSION_MAJOR=`echo "${REQU_VERSION_SAFE}" | tr  '.' ' ' | awk '{print $1}'`
+    REQU_VERSION_MINOR=`echo "${REQU_VERSION_SAFE}" | tr  '.' ' ' | awk '{print $2}'`
+    REQU_VERSION_PATCH=`echo "${REQU_VERSION_SAFE}" | tr  '.' ' ' | awk '{print $3}'`
+
+    if [ -z "${CURR_VERSION_MAJOR}" ] ||
+	[ -z "${CURR_VERSION_MINOR}" ] ||
+	[ -z "${CURR_VERSION_PATCH}" ] ||
+	[ -z "${REQU_VERSION_MAJOR}" ] ||
+	[ -z "${REQU_VERSION_MINOR}" ] ||
+	[ -z "${REQU_VERSION_PATCH}" ]; then
+	VERSION_INVALID=1
+	return
+    fi
+
+    if [ ${CURR_VERSION_MAJOR} -lt ${REQU_VERSION_MAJOR} ]; then
+	VERSION_INVALID=1
+	return
+    else
+	if [ ${CURR_VERSION_MINOR} -lt ${REQU_VERSION_MINOR} ]; then
+	    VERSION_INVALID=1
+	    return
+	else
+	    if [ ${CURR_VERSION_PATCH} -lt ${REQU_VERSION_PATCH} ]; then
+		VERSION_INVALID=1
+		return
+	    fi
+	fi
+    fi
+}
+
+checkMinVersionCompatibility "${AMP_APP_VERSION_CURRENT}" "${AMP_APP_VERSION_MINIMUM}"
+AMP_APP_VERSION_INVALID=${VERSION_INVALID}
+
+checkMinVersionCompatibility "${AMP_CLI_VERSION_CURRENT}" "${AMP_CLI_VERSION_MINIMUM}"
+AMP_CLI_VERSION_INVALID=${VERSION_INVALID}
+
+if [ ${AMP_CLI_VERSION_INVALID} -eq 1 ]; then
+    echo "ERROR: Minimum version required of Amplify CLI is not installed."
```

Comment:
Make this lowercase (`echo "error: ..."`) so it will show up in the main interface Issue Navigator, instead of making the customer go to the build logs to inspect an arcane "Run script build phase exited with nonzero exit status" message.